python-package/lightgbm/engine.py

```diff
@@ -198,11 +254,10 @@ def train(
     if callable(params["objective"]):
         fobj = params["objective"]
         params["objective"] = "none"
-    for alias in _ConfigAliases.get("num_iterations"):
-        if alias in params:
-            num_boost_round = params.pop(alias)
-            _log_warning(f"Found `{alias}` in params. Will use it instead of argument")
-    params["num_iterations"] = num_boost_round
+
+    params = _choose_num_iterations(num_boost_round_kwarg=num_boost_round, params=params)
+    num_boost_round = params["num_iterations"]
+
```

Comment:
I think the following line should be moved here now
https://github.com/microsoft/LightGBM/blob/63e15300ec88c143f57e1214d10e3d6aa87b92c4/python-package/lightgbm/engine.py#L228-L229

Reply:
Agreed! I think maybe it always should have been after this processing, to prevent **any** values `<= 0` from making it into `"num_iterations"` being passed to the C++ side.

Reply:
updated in https://github.com/microsoft/LightGBM/pull/6579/commits/b6b5c806c840640b348fe373aa4255fd83bce4b5